packages/cli/src/evaluation.ee/test-runner/__tests__/test-runner.service.ee.test.ts

```diff
@@ -753,6 +754,120 @@ describe('TestRunnerService', () => {
 		});
 	});
 
+	test('should create proper execution data for queue mode in runTestCase', async () => {
+		config.set('executions.mode', 'queue');
+
+		const testRunnerService = new TestRunnerService(
+			logger,
+			telemetry,
+			workflowRepository,
+			workflowRunner,
+			executionRepository,
+			activeExecutions,
+			testRunRepository,
+			testCaseExecutionRepository,
+			testMetricRepository,
+			mockNodeTypes,
+			errorReporter,
+		);
+
+		// Spy on workflowRunner.run to capture the data passed to it
+		jest.spyOn(workflowRunner, 'run').mockImplementation(async (data) => {
+			// Verify the data structure is correct for queue mode
+			expect(data.executionMode).toBe('evaluation');
+
+			// Check that executionData field is properly defined
+			expect(data.executionData).toBeDefined();
+			expect(data.executionData!.startData).toBeDefined();
+			expect(data.executionData!.startData!.startNodes).toBeDefined();
+			expect(data.executionData!.resultData.pinData).toBeDefined();
+			expect(data.executionData!.resultData.runData).toBeDefined();
+			expect(data.executionData!.manualData!.userId).toBeDefined();
+			expect(data.executionData!.manualData!.partialExecutionVersion).toBe(2);
+			expect(data.executionData!.manualData!.triggerToStartFrom).toBeDefined();
+
+			return 'mock-execution-id';
+		});
+
+		// Mock activeExecutions.getPostExecutePromise to return a successful execution
+		activeExecutions.getPostExecutePromise.mockResolvedValue(mockExecutionData());
+
+		// Create an AbortController for the test
+		const abortController = new AbortController();
+
+		// Setup test metadata
+		const metadata: any = {
+			testRunId: 'test-run-id',
+			userId: 'user-id',
+			pastExecutionId: 'past-execution-id',
+		};
+
+		// Call runTestCase directly to test the executionData construction
+		await (testRunnerService as any).runTestCase(
```

Comment:
Nit: You wrote the service, so you know best, but I'd say it would be better if we could avoid calling private functions in tests and rely on the public interface.